### DISCUSSION TITLE
What types of new control data can I add?

### DISCUSSION THREAD

**Discussion by anxcon** [lllyasviel/ControlNet#286] [Q&A]
What types of new control data can I add through ControlNet? Example can I train a linear/ranged value for height/weight of a person, or joint angle, and it will understand anything along that range?

**Comment by Njasa2k** [comment]
Text detection -> text generation? Facial expression recognition -> facial expression generation? Gesture recognition -> gesture generation? color detection -> color generation? texture detection -> texture generation? Object recognition -> object generation? style detection -> style generation? facial recognition to generation?

maybe anything that can be detected can be used for training a controlnet model? I'm not really sure.

**Comment by anxcon** [comment]
What I would like is to input an image, and a few 0.0-1.0 values with it. Then later input something in their range to aid in generation. Example say weight, training from images and values for everything from 100 pounds to 200 pounds assigned as 0.0-1.0. Then generate an image using 0.3 or 0.79. Is this possible with ControlNet?

**Reply by Njasa2k** [reply]
I think that would be difficult for a text to image generator to get the exact appearance of a person with a certain weight or height. Is there a type of detector that can detect a person's height and weight from just the appearance.

**Reply by anxcon** [reply]
Yes img2weight and img2height exist, but I feel like that's getting ahead of things. The question is more about if ControlNet is even capable of training off of a value in that way. Could train it to know the RGB value to paint a car, how long a bridge is, length of hair, etc.

**Comment by jarry-LU** [comment]
I think, can you add a new type of ControlNet process about the clothes type ?
In many times, when we create a character even thought we had use some lora.
But we still can not Control the clothes type precise.
If stable diffusion can control the clothes type,It will be used in many place.
我英文不好,因此附上中文
我想說,如果可以的話,您可以考慮增加指定[衣物]的處裡模式
如此一來stable diffusion就可以更具體的被使用在更多的地方
且生成的圖片將更具有一致姓. 也能用於 衣物穿搭的模擬. 或是提供給時裝設計師一個很好的參考~
很多像我一樣的AI生成使用者都很期盼這個功能,這無疑是AI生成的[聖杯]. 新的奇異點!
